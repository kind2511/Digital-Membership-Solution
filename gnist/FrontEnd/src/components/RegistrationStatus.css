.registration-status {
  display: flex;
  align-items: center;
  cursor: pointer;
  font-weight: bold;
  font-size: 20px;
}

.register-checkbox {
  margin-right: 5px;
}

.registration-modal-overlay {
  position: fixed;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  background-color: rgba(0, 0, 0, 0.7);
  display: flex;
  justify-content: center;
  align-items: center;
  z-index: 1000;
}

.registration-modal-content {
  background: #fff;
  padding: 20px;
  border-radius: 10px;
  width: 300px;
  text-align: center;
}

.registration-modal-content button {
  padding: 8px 16px;
  margin: 0 8px;
  cursor: pointer;
  background-color: #4caf50;
  color: white;
  border: none;
  border-radius: 5px;
  margin-top: 10px;
}

.registration-modal-content button:nth-child(2) {
  background-color: #f44336;
}

.registration-modal-content button:hover {
  opacity: 0.9;
}
