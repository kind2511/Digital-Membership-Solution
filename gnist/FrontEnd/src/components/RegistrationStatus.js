import React, { useEffect, useState } from 'react';
import axios from 'axios';
import './RegistrationStatus.css';

function RegistrationStatus({ userSub, isRegistered, setIsRegistered }) {
  const [showConfirmModal, setShowConfirmModal] = useState(false);
  const [confirmationMessage, setConfirmationMessage] = useState('');
  const baseApiUrl = 'http://localhost:8000';

  useEffect(() => {
    if (userSub) {
      const storedRegistrationStatus = localStorage.getItem('isRegistered');
      const isUserRegistered = storedRegistrationStatus ? storedRegistrationStatus === 'true' : false;
      setIsRegistered(isUserRegistered);
      setConfirmationMessage(isUserRegistered ? 'Du er registrert nå.' : 'Du er ikke registrert nå.');
    }
  }, [userSub, setIsRegistered]);


  const handleCheckboxClick = () => {
    setShowConfirmModal(true);
  };

  const handleRegistrationChange = async (confirm) => {
    if (confirm) {
      const endpoint = `${baseApiUrl}/digital_medlemsordning/add_day/${userSub}/`;
      try {
        await axios.get(endpoint);
        const newRegistrationStatus = !isRegistered;
        setIsRegistered(newRegistrationStatus);
        localStorage.setItem('isRegistered', newRegistrationStatus);
        setConfirmationMessage(newRegistrationStatus ? 'Du er registrert nå.' : 'Du er ikke registrert nå.');
      } catch (error) {
        console.error("Error toggling registration:", error.response || error);
      }
    }
    setShowConfirmModal(false);
  };

  return (
    <div className="registration-status">
      <input
        type="checkbox"
        checked={isRegistered}
        onChange={() => { }}
        onClick={handleCheckboxClick}
        className="register-checkbox"
      />
      Registrer
      {showConfirmModal && (
        <div className="registration-modal-overlay">
          <div className="registration-modal-content">
            <p>{isRegistered ? 'Er du sikker på at du vil avregistrere deg?' : 'Er du sikker på at du vil registrere deg?'}</p>
            <button onClick={() => handleRegistrationChange(true)}>Ja</button>
            <button onClick={() => handleRegistrationChange(false)}>Nei</button>
          </div>
        </div>
      )}
      {confirmationMessage && (
        <div className="confirmation-message">{confirmationMessage}</div>
      )}
    </div>
  );
}

export default RegistrationStatus;
