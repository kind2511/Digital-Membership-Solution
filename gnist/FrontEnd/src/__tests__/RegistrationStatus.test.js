import React from 'react';
import renderer from 'react-test-renderer';
import RegistrationStatus from '../components/RegistrationStatus';
import axios from 'axios';
import MockAdapter from 'axios-mock-adapter';

// Mock axios for the test
const mock = new MockAdapter(axios);

const baseApiUrl = 'http://localhost:8000';
const userSub = 'auth0|123456789';
const endpoint = `${baseApiUrl}/digital_medlemsordning/add_day/${userSub}/`;

mock.onPost(endpoint).reply(200);

const defaultProps = {
  userSub,
  isRegistered: false,
  setIsRegistered: jest.fn(),
};

const createTree = (props) => {
  return renderer.create(<RegistrationStatus {...props} />).toJSON();
};

test('RegistrationStatus component snapshot - default state', () => {
  const tree = createTree(defaultProps);
  expect(tree).toMatchSnapshot();
});

test('RegistrationStatus component snapshot - registered state', () => {
  const tree = createTree({ ...defaultProps, isRegistered: true });
  expect(tree).toMatchSnapshot();
});

test('RegistrationStatus component snapshot - with confirmation modal', () => {
  const component = renderer.create(<RegistrationStatus {...defaultProps} />);
  const instance = component.root;

  // Checkbox click
  const checkbox = instance.findByType('input');
  checkbox.props.onClick();

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
