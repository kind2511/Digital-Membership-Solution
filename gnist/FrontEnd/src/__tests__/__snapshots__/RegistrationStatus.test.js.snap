// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegistrationStatus component snapshot - default state 1`] = `
<div
  className="registration-status"
>
  <label
    className="switch"
  >
    <input
      checked={false}
      onChange={[Function]}
      onClick={[Function]}
      type="checkbox"
    />
    <span
      className="slider round"
    />
  </label>
  <span
    className="status-text"
  >
    Ikke Aktiv
  </span>
</div>
`;

exports[`RegistrationStatus component snapshot - registered state 1`] = `
<div
  className="registration-status"
>
  <label
    className="switch"
  >
    <input
      checked={true}
      onChange={[Function]}
      onClick={[Function]}
      type="checkbox"
    />
    <span
      className="slider round"
    />
  </label>
  <span
    className="status-text"
  >
    Aktiv
  </span>
</div>
`;

exports[`RegistrationStatus component snapshot - with confirmation modal 1`] = `
<div
  className="registration-status"
>
  <label
    className="switch"
  >
    <input
      checked={false}
      onChange={[Function]}
      onClick={[Function]}
      type="checkbox"
    />
    <span
      className="slider round"
    />
  </label>
  <span
    className="status-text"
  >
    Ikke Aktiv
  </span>
  <div
    className="registration-modal-overlay"
  >
    <div
      className="registration-modal-content"
    >
      <p>
        Er du sikker p√• at du vil registrere deg?
      </p>
      <button
        onClick={[Function]}
      >
        Ja
      </button>
      <button
        onClick={[Function]}
      >
        Nei
      </button>
    </div>
  </div>
</div>
`;
